### PR TITLE
Fix multiple broadcast for cross txs

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added / changed some prints and their log level in broadcast component
* Added a missed return

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: consensus/broadcast/shardChainMessenger.go [hunk 1]
```diff
@@ -231,6 +232,7 @@ func (scm *shardChainMessenger) PrepareBroadcastBlockDataValidator(
 	err = scm.delayedBlockBroadcaster.SetValidatorData(broadcastData)
 	if err != nil {
 		log.Error("shardChainMessenger.PrepareBroadcastBlockDataValidator", "error", err)
+		return
```
Comment (iulianpascalau) [inline consensus/broadcast/shardChainMessenger.go:235]: :+1:

**Review by iulianpascalau — APPROVED** [review]

File: consensus/broadcast/shardChainMessenger.go (hunk 1 above)
Comment (AdoAdoAdo) [inline consensus/broadcast/shardChainMessenger.go:235]: this return is redundant

File: consensus/broadcast/shardChainMessenger.go [hunk 2]
```diff
@@ -196,6 +196,7 @@ func (scm *shardChainMessenger) PrepareBroadcastHeaderValidator(
 	err = scm.delayedBlockBroadcaster.SetHeaderForValidator(vData)
 	if err != nil {
 		log.Error("shardChainMessenger.PrepareBroadcastHeaderValidator", "error", err)
+		return
```
Comment (AdoAdoAdo) [inline consensus/broadcast/shardChainMessenger.go:199]: this return is redundant

File: consensus/broadcast/metaChainMessenger.go [hunk 3]
```diff
@@ -160,6 +161,7 @@ func (mcm *metaChainMessenger) PrepareBroadcastHeaderValidator(
 	err = mcm.delayedBlockBroadcaster.SetHeaderForValidator(vData)
 	if err != nil {
 		log.Error("metaChainMessenger.PrepareBroadcastHeaderValidator", "error", err)
+		return
```
Comment (AdoAdoAdo) [inline consensus/broadcast/metaChainMessenger.go:164]: this return is redundant

File: consensus/broadcast/metaChainMessenger.go [hunk 4]
```diff
@@ -141,6 +141,7 @@ func (mcm *metaChainMessenger) PrepareBroadcastHeaderValidator(
 ) {
 	if check.IfNil(header) {
 		log.Error("metaChainMessenger.PrepareBroadcastHeaderValidator", "error", spos.ErrNilHeader)
+		return
```
Comment (AdoAdoAdo) [inline consensus/broadcast/metaChainMessenger.go:144]: :+1:

File: consensus/broadcast/shardChainMessenger.go (hunk 1 above)
Comment (SebastianMarian) [reply consensus/broadcast/shardChainMessenger.go:235]: I would keep it just for the case when the method would be extended by someone who would forgot to added back

File: consensus/broadcast/shardChainMessenger.go (hunk 2 above)
Comment (SebastianMarian) [reply consensus/broadcast/shardChainMessenger.go:199]: I would keep it just for the case when the method would be extended by someone who would forgot to added back

File: consensus/broadcast/metaChainMessenger.go (hunk 3 above)
Comment (SebastianMarian) [reply consensus/broadcast/metaChainMessenger.go:164]: I would keep it just for the case when the method would be extended by someone who would forgot to added back

**Review by AdoAdoAdo — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**